src/main/java/io/strimzi/kproxy/codec/DecodedFrame.java

```diff
@@ -113,4 +119,16 @@ public String toString() {
                 ", body=" + body +
                 ')';
     }
+
+    @Override
+    public void add(ByteBuf buffer) {
+        buffers.add(buffer);
+    }
+
+    @Override
+    public void release() {
+        for (ByteBuf buffer : buffers) {
+            buffer.release();
```

Comment:
What happen to the held buffers? 
they are supposed to be removed from the collection as well?

Reply:
I could remove them, but I'd expect Java GC to take care of it when collecting the `DecodedFrame` object itself?

Reply:
Yep, fair enough :+1: 
see https://github.com/kroxylicous/kproxy/pull/7#discussion_r864756368 for a licecycle related comment

Reply:
Yepp, still processing that one :) I'll mark this one here as resolved then.